test/cases/enum.ts

```diff
@@ -6,7 +6,8 @@ export var schema = {
       "enum": ["a", "b", "c"]
     },
     "bar": {
-      "enum": [1, 2, 3]
+      "enum": [1, 2, 3],
+      "tsEnumNames": ["One","Two","Three"]
```

Comment:
> Exactly.

In that case, why not pre-process the schema before piping it into json-schema-to-ts?

> But I think more importantly, the point of json-schema is to declare valid values for what goes over the wire, and in this case for an integer-based enum the values that go over the wire will be integers

I agree. In TS, the way to do that is to specify a mapping from String => Number (via, for example, tsEnumValues). So it is a bit awkward to send integers when the JSON Schema says that values are integers, but that's what the tsEnumValues property helps with.

In any case, I think a pre processing step would be the cleanest solution here. What do you think?

In the future, if TS implements [complex enum types](https://github.com/Microsoft/TypeScript/issues/1206) that can make this easier. Also, [number literal types](https://github.com/Microsoft/TypeScript/pull/9407) might be another good solution.